STMicroelectronics/SPWF04Sx/SPWF04SxInterface.cs

```diff
@@ -187,6 +187,418 @@ public class SPWF04SxInterface : NetworkInterface, ISocketProvider, ISslStreamPr
             return result.Substring(result.IndexOf(':') + 1);
         }
 
+        //*************************************** Begin Commands added by RoSchmi  ***************************************************
+
+
+        /// <summary>
+        /// Computes a hash
+        /// 0:SHA1, 1:SHA224, 2:SHA256, 3:MD5
+        /// </summary>
+        /// <param name="function">The methode to use. 0:SHA1, 1:SHA224, 2:SHA256, 3:MD5</param>
+        /// <param name="filename">The name of the file that holds the data
+        public byte[] ComputeHash(string function, string filename)   // 0:SHA1, 1:SHA224, 2:SHA256, 3:MD5
+        {
+            var cmd = this.GetCommand()
+                .AddParameter(function)
+                .AddParameter(filename)
+                .Finalize(SPWF04SxCommandIds.HASH);
+            this.EnqueueCommand(cmd);
+            byte[] totalBuf = new byte[0];
```

Comment:
This type of logic appears to be used in multiple places, to read a buffer. Perhaps it's possible to abstract it to a common private helper function?